src/main/scala/io/computenode/cyfra/dsl/GSeq.scala

```diff
@@ -119,12 +117,12 @@ object GSeq:
     val streamNextExpr = seq.source.next 
     val seqExprs = seq.elemOps.map(_.fn)
     
-    val limitExpr = ConstInt32(seq.limit.getOrElse(throw new IllegalArgumentException("Reduce on infinite stream is not supported")))
+    // Ensure a default limit to avoid infinite stream errors
+    val defaultLimit = 100 // Set a safe default limit
```

Comment:
I don't think it is a good idea. It's pretty similar to letting people write fors in a language without upper bound, and having default = 100, it can be quite confusing. If you would like to make it safer you could make a `GSeq` reduce result only be a `Value` after limit is introduced. That would be quite challanging but a good Scala excercise.